pkg/check/manifest/manifest.go

```diff
@@ -86,57 +105,169 @@ func (c *Check) Run(ctx context.Context, cluster orchestration.Cluster, opts int
+	if err != nil {
+		return err
+	}
+	tarFile := bee.NewBufferFile("", tarReader)
+	if err := upClient.UploadCollection(ctx, &tarFile, api.UploadOptions{BatchID: batchID, IndexDocument: "index.html"}); err != nil {
+		return err
+	}
+	c.logger.Infof("collection uploaded: %s", tarFile.Address())
+	time.Sleep(3 * time.Second)
+
+	// push first version of website to the feed
+	ref, err := upClient.UpdateFeedWithReference(ctx, signer, topic, 0, tarFile.Address(), api.UploadOptions{BatchID: batchID})
+	if err != nil {
+		return err
+	}
+	c.logger.Infof("feed updated: %s", ref.Reference)
+
+	// download root (index.html) from the feed
+	err = c.download(downClient, rootFeedRef.Reference, nil, files[0])
+	if err != nil {
+		return err
+	}
+
+	// update  website files
+	files, err = generateFilesWithPaths(rnd, paths, int(o.MaxPathnameLength))
+	if err != nil {
+		return err
 	}
 
-	lastNode := sortedNodes[len(sortedNodes)-1]
-	try := 0
+	tarReader, err = tarFiles(files)
+	if err != nil {
+		return err
+	}
+	tarFile = bee.NewBufferFile("", tarReader)
+	if err := upClient.UploadCollection(ctx, &tarFile, api.UploadOptions{BatchID: batchID, IndexDocument: "index.html"}); err != nil {
+		return err
+	}
+	time.Sleep(3 * time.Second)
 
-DOWNLOAD:
-	time.Sleep(5 * time.Second)
-	try++
-	if try > 5 {
-		return errors.New("failed getting manifest files after too many retries")
+	// push 2nd version of website to the feed
+	ref, err = upClient.UpdateFeedWithReference(ctx, signer, topic, 1, tarFile.Address(), api.UploadOptions{BatchID: batchID})
+	if err != nil {
+		return err
 	}
+	c.logger.Infof("feed updated: %s", ref.Reference)
 
-	for i, file := range files {
-		node := clients[lastNode]
+	// download updated index.html from the feed
+	err = c.download(downClient, rootFeedRef.Reference, nil, files[0])
+	if err != nil {
+		return err
+	}
 
-		size, hash, err := node.DownloadManifestFile(ctx, tarFile.Address(), file.Name())
+	// download other paths and compare
+	for i := 0; i < len(files); i++ {
+		err = c.download(downClient, tarFile.Address(), &files[i], files[0])
 		if err != nil {
-			c.logger.Infof("Node %s. Error retrieving file: %v", lastNode, err)
-			goto DOWNLOAD
+			return err
 		}
+	}
+	return nil
+}
 
-		if !bytes.Equal(file.Hash(), hash) {
-			c.logger.Infof("Node %s. File %d not retrieved successfully. Uploaded size: %d Downloaded size: %d Node: %s File: %s/%s", lastNode, i, file.Size(), size, overlays[lastNode].String(), tarFile.Address().String(), file.Name())
-			return errManifest
+// download retrieves a file from the given address using the specified client.
+// If the file parameter is nil, it downloads the index file in the collection.
+func (c *Check) download(client *bee.Client, address swarm.Address, file *bee.File, indexFile bee.File) error {
+	fName := ""
+	if file != nil {
+		fName = file.Name()
+	}
+	c.logger.Infof("downloading file: %s/%s", address, fName)
+
+	var hash []byte
+	for i := 0; i < 5; i++ {
```

Comment:
is some many retries necessary? is this just a precaution or you saw that it was failing quite often?

Reply:
The retry count was always 5. I just refactored the code. I didn't want to mess with it.
https://github.com/ethersphere/beekeeper/blob/master/pkg/check/manifest/manifest.go#L115